### ISSUE TITLE
Warnings regarding unused osmajorrelease grain

### ISSUE THREAD

**Issue by boltronics** [saltstack/salt#36359]
### Description of Issue/Question
After upgrading to 2016.3.3, I get the following warning on each `salt-call` execution:
```
[WARNING ] /usr/lib/python2.7/dist-packages/salt/grains/core.py:1493: DeprecationWarning: The "osmajorrelease" will be a type of an integer.
```
After running a recursive grep through all of my state and pillar files, I have come to the conclusion that I'm not using this anywhere. If I'm not using it, I shouldn't be warned about it.

In my opinion, warnings tell you that you are doing something wrong, and there could be a potential problem or a better way to do something. Things not applicable to my setup should not produce warnings, and documenting the change in release notes would be more appropriate.

Each time I see a warning, I try to be proactive and correct it. It's a waste of time and mental energy to investigate warnings on each execution that AFAICT aren't applicable to me.

### Setup
I suspect the warning will appear regardless of the states in use.

### Steps to Reproduce Issue
Use `salt-call` to do basically anything.

### Versions Report
```
# apt-cache policy salt-minion | head -n 3
salt-minion:
  Installed: 2016.3.3-1
  Candidate: 2016.3.3-1

# salt-call --versions-report
Salt Version:
           Salt: 2016.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.0
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.1
            ZMQ: 4.0.5
 
System Versions:
           dist: debian 8.5 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.5 
 ```
The salt-minion package is built using the `v2016.3.3` tagged git commit (which was tagged incorrectly again since it did not include the version bump). That's a different issue.

**Comment by gtmanfred** [comment]
Thanks for the report, I have been seeing this as well.

**Comment by cachedout** [comment]
This should be fixed. Please check the head of the 2016.3 branch.

**Comment by boltronics** [comment]
Nice. Thanks!

It's Saturday 1am here and I've got a busy weekend ahead of me, so I will try to look at this on Monday.

**Comment by rallytime** [comment]
@boltronics Yeah, this was fixed by #36068. :)

**Comment by boltronics** [comment]
Cool. Looks like it was fixed days earlier. Could have sworn I searched for existing issues regarding `osmajorrelease` but must have missed it. Sorry for the noise.
